### ISSUE TITLE
CustomLearningAdmin.aspx blank after provisioning

### ISSUE THREAD

**Issue by MichaelLaMontagne** [pnp/custom-learning-office-365#10]
After receiving the provisioning is complete email, navigating to my custom learning url//SitePages/CustomLearningAdmin.aspx the page loads the title and search bar for a second and after just loads a blank page.

**Comment by karuanag** [comment]
Thanx for this Michael.  Which browser are you working within?

**Comment by MichaelLaMontagne** [comment]
Google Chrome: Version 73.0.3683.75 (Official Build) (64-bit)

**Comment by hmank** [comment]
I'm having the same issue, I've tried on the following browsers:
Chrome Version 72.0.3626.121 (Official Build) (64-bit)
Microsoft Edge 44.17763.1.0, Microsoft EdgeHTML 18.17763

**Comment by gmcanallymcny** [comment]
I'm having this same issue for 
Chrome Version 72.0.3626.121 (Official Build) (64-bit)

I am able to see content in Edge and IE, but these are not the browsers regularly used by my user-base.

**Comment by juliemturner** [comment]
#11

**Comment by juliemturner** [comment]
Duplicate of #11

**Comment by juliemturner** [comment]
We've been able to replicate this issue when the tenant is new and the app catalog was created but 30 minutes didn't elapse before trying to provision using the provisioning service.  For those reporting this issue, please confirm by retrying provisioning to your tenant and see if that corrects the issue.

**Comment by hmank** [comment]
Hi Julie, I have had this Tenant for 3 years. I just deleted the O365CL sharepoint site and also removed from deleted sites. I then redeployed and it was completed successfully, however I still have the same issue with the blank CustomLearningAdmin.aspx page.
![O365clerr](https://user-images.githubusercontent.com/32819512/54373555-d4ec5a00-4653-11e9-8a87-98aa7904efd2.jpg)

**Comment by juliemturner** [comment]
Can you navigate to the CustomLearningViewer.aspx page, and does that work successfully (I'm going to assume no). If you just created the app catalog site collection that would be the same problem even if the tenant isn't new.  The issue is the webparts package cannot be properly deployed in a "new" app catalog site collection without giving it time to "percolate"

**Comment by hmank** [comment]
Same issue for the CustomLearningViewer.aspx page, is there something I can do the help with the "percolation"?

**Comment by juliemturner** [comment]
Basically your site was not provisioned properly.  The web parts are not registered properly and so they won't load in the page. If you've waited the 30 minutes since the app catalog was created and tried reprovisioning then I would say you have two options. 

1. Remove the site collection and whatever app might have ended up in the app catalog and try again. If it still doesn't work I would submit an issue with pnp provisioning (https://github.com/SharePoint/sp-dev-docs/issues). 

2. You can also get a copy of the sppkg file from this repo and manually try uploading that to your app catalog and see if that will fix the issue.

**Comment by hmank** [comment]
Hi Julie, so I finally got it to work here are the steps performed:

1. Removed the Custom Learning Site Collection
2. Removed the appcatalog site collection
3. Recreated the appcatalog site collection
4. Redeployed the custom learning package
5. I received an email stating :

Unfortunately, your request to provision"Custom learning for Office 365" template failed. | Unfortunately, your request to provision"Custom learning for Office 365" template failed.
-- | --
Unfortunately, your request to provision"Custom learning for Office 365" template failed.
 
Here is the exception that occurred:Type = System.Net.WebException Message = The remote server returned an error: (404) Not Found. Source = System Status = ProtocolError Response = System.Net.HttpWebResponse Data = [0] = ClientRequest : Microsoft.SharePoint.Client.ClientRequestWrapper TargetSite = System.Net.WebResponse EndGetResponse(System.IAsyncResult) StackTrace = at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult) at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.SPWebRequestExecutor.<ExecuteAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryToServerAsync>d__6.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRuntimeContext.<ExecuteQueryAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContext.<ExecuteQueryAsync>d__4.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext() --- End of stack trace from previous location where exception was thrown --- at Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryRetry(ClientRuntimeContext clientContext, Int32 retryCount, Int32 delay, String userAgent) at Microsoft.SharePoint.Client.NavigationExtensions.AddNavigationNode(Web web, String nodeTitle, Uri nodeUri, String parentNodeTitle, NavigationType navigationType, Boolean isExternal, Boolean asLastNode, String l1ParentNodeTitle) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionStructuralNavigationNodes(Web web, TokenParser parser, NavigationType navigationType, NavigationNodeCollection nodes, PnPMonitoredScope scope, String parentNodeTitle, String l1ParentNodeTitle) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionStructuralNavigation(Web web, StructuralNavigation structuralNavigation, TokenParser parser, Boolean currentNavigation, Boolean clearNavigation, PnPMonitoredScope scope) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionCurrentStructuralNavigation(Web web, StructuralNavigation structuralNavigation, TokenParser parser, Boolean clearNavigation, PnPMonitoredScope scope) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionObjects(Web web, ProvisioningTemplate template, TokenParser parser, ProvisioningTemplateApplyingInformation applyingInformation) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.ApplyRemoteTemplate(Web web, ProvisioningTemplate template, ProvisioningTemplateApplyingInformation provisioningInfo, Boolean calledFromHierarchy, TokenParser tokenParser) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectHierarchySequenceSites.ProvisionObjects(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, TokenParser tokenParser, ProvisioningTemplateApplyingInformation applyingInformation) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.ApplyProvisioningHierarchy(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, ProvisioningTemplateApplyingInformation provisioningInfo) at Microsoft.SharePoint.Client.TenantExtensions.ApplyProvisionHierarchy(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, ProvisioningTemplateApplyingInformation applyingInformation) at SharePointPnP.ProvisioningApp.WebJob.ProvisioningFunction.<RunAsync>d__0.MoveNext() in C:\github\sp-provisioning-service\SharePointPnP.ProvisioningApp\SharePointPnP.ProvisioningApp.WebJob\ProvisioningFunction.cs:line 337 HResult = -2146233079 | Here is the exception that occurred:Type = System.Net.WebException Message = The remote server returned an error: (404) Not Found. Source = System Status = ProtocolError Response = System.Net.HttpWebResponse Data = [0] = ClientRequest : Microsoft.SharePoint.Client.ClientRequestWrapper TargetSite = System.Net.WebResponse EndGetResponse(System.IAsyncResult) StackTrace = at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult) at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.SPWebRequestExecutor.<ExecuteAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryToServerAsync>d__6.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRuntimeContext.<ExecuteQueryAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContext.<ExecuteQueryAsync>d__4.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext() --- End of stack trace from previous location where exception was thrown --- at Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryRetry(ClientRuntimeContext clientContext, Int32 retryCount, Int32 delay, String userAgent) at Microsoft.SharePoint.Client.NavigationExtensions.AddNavigationNode(Web web, String nodeTitle, Uri nodeUri, String parentNodeTitle, NavigationType navigationType, Boolean isExternal, Boolean asLastNode, String l1ParentNodeTitle) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionStructuralNavigationNodes(Web web, TokenParser parser, NavigationType navigationType, NavigationNodeCollection nodes, PnPMonitoredScope scope, String parentNodeTitle, String l1ParentNodeTitle) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionStructuralNavigation(Web web, StructuralNavigation structuralNavigation, TokenParser parser, Boolean currentNavigation, Boolean clearNavigation, PnPMonitoredScope scope) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionCurrentStructuralNavigation(Web web, StructuralNavigation structuralNavigation, TokenParser parser, Boolean clearNavigation, PnPMonitoredScope scope) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionObjects(Web web, ProvisioningTemplate template, TokenParser parser, ProvisioningTemplateApplyingInformation applyingInformation) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.ApplyRemoteTemplate(Web web, ProvisioningTemplate template, ProvisioningTemplateApplyingInformation provisioningInfo, Boolean calledFromHierarchy, TokenParser tokenParser) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectHierarchySequenceSites.ProvisionObjects(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, TokenParser tokenParser, ProvisioningTemplateApplyingInformation applyingInformation) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.ApplyProvisioningHierarchy(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, ProvisioningTemplateApplyingInformation provisioningInfo) at Microsoft.SharePoint.Client.TenantExtensions.ApplyProvisionHierarchy(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, ProvisioningTemplateApplyingInformation applyingInformation) at SharePointPnP.ProvisioningApp.WebJob.ProvisioningFunction.<RunAsync>d__0.MoveNext() in C:\github\sp-provisioning-service\SharePointPnP.ProvisioningApp\SharePointPnP.ProvisioningApp.WebJob\ProvisioningFunction.cs:line 337 HResult = -2146233079
Here is the exception that occurred:Type = System.Net.WebException Message = The remote server returned an error: (404) Not Found. Source = System Status = ProtocolError Response = System.Net.HttpWebResponse Data = [0] = ClientRequest : Microsoft.SharePoint.Client.ClientRequestWrapper TargetSite = System.Net.WebResponse EndGetResponse(System.IAsyncResult) StackTrace = at System.Net.HttpWebRequest.EndGetResponse(IAsyncResult asyncResult) at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization) --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.SPWebRequestExecutor.<ExecuteAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryToServerAsync>d__6.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRequest.<ExecuteQueryAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientRuntimeContext.<ExecuteQueryAsync>d__0.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContext.<ExecuteQueryAsync>d__4.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext() --- End of stack trace from previous location where exception was thrown --- at Microsoft.SharePoint.Client.ClientContextExtensions.<ExecuteQueryImplementation>d__7.MoveNext() --- End of stack trace from previous location where exception was thrown --- at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task) at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task) at Microsoft.SharePoint.Client.ClientContextExtensions.ExecuteQueryRetry(ClientRuntimeContext clientContext, Int32 retryCount, Int32 delay, String userAgent) at Microsoft.SharePoint.Client.NavigationExtensions.AddNavigationNode(Web web, String nodeTitle, Uri nodeUri, String parentNodeTitle, NavigationType navigationType, Boolean isExternal, Boolean asLastNode, String l1ParentNodeTitle) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionStructuralNavigationNodes(Web web, TokenParser parser, NavigationType navigationType, NavigationNodeCollection nodes, PnPMonitoredScope scope, String parentNodeTitle, String l1ParentNodeTitle) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionStructuralNavigation(Web web, StructuralNavigation structuralNavigation, TokenParser parser, Boolean currentNavigation, Boolean clearNavigation, PnPMonitoredScope scope) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionCurrentStructuralNavigation(Web web, StructuralNavigation structuralNavigation, TokenParser parser, Boolean clearNavigation, PnPMonitoredScope scope) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectNavigation.ProvisionObjects(Web web, ProvisioningTemplate template, TokenParser parser, ProvisioningTemplateApplyingInformation applyingInformation) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.ApplyRemoteTemplate(Web web, ProvisioningTemplate template, ProvisioningTemplateApplyingInformation provisioningInfo, Boolean calledFromHierarchy, TokenParser tokenParser) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.ObjectHierarchySequenceSites.ProvisionObjects(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, TokenParser tokenParser, ProvisioningTemplateApplyingInformation applyingInformation) at OfficeDevPnP.Core.Framework.Provisioning.ObjectHandlers.SiteToTemplateConversion.ApplyProvisioningHierarchy(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, ProvisioningTemplateApplyingInformation provisioningInfo) at Microsoft.SharePoint.Client.TenantExtensions.ApplyProvisionHierarchy(Tenant tenant, ProvisioningHierarchy hierarchy, String sequenceId, ProvisioningTemplateApplyingInformation applyingInformation) at SharePointPnP.ProvisioningApp.WebJob.ProvisioningFunction.<RunAsync>d__0.MoveNext() in C:\github\sp-provisioning-service\SharePointPnP.ProvisioningApp\SharePointPnP.ProvisioningApp.WebJob\ProvisioningFunction.cs:line 337 HResult = -2146233079

6. 6 minutes later I received another email : 

Your request to provision"Custom learning for Office 365" template is now completed! | Your request to provision"Custom learning for Office 365" template is now completed!
-- | --
Your request to provision"Custom learning for Office 365" template is now completed!
 
I was able to browse the Custom Learning Site and browse the Admin Page.

Habib
